interfaces/sourcegen/sourcegen/yaml/_YamlSourceGenerator.py

```diff
@@ -0,0 +1,66 @@
+"""
+Generator for YAML output.
+
+Used for illustration purposes: the `CLibSourceGenerator` is used to preprocess YAML
+header specifications, which yields the `HeaderFile` objects used by this source
+generator.
+"""
+
+# This file is part of Cantera. See License.txt in the top-level directory or
+# at https://cantera.org/license.txt for license and copyright information.
+
+from pathlib import Path
+import logging
+
+from jinja2 import Environment, BaseLoader
+
+from .._dataclasses import HeaderFile, CFunc
+from .._SourceGenerator import SourceGenerator
+
+
+_LOGGER = logging.getLogger()
+
+class YamlSourceGenerator(SourceGenerator):
+    """The SourceGenerator for generating CLib."""
+
+    def __init__(self, out_dir: str, config: dict, templates: dict):
+        self._out_dir = out_dir or None
+        if self._out_dir is not None:
+            self._out_dir = Path(out_dir)
+            self._out_dir.mkdir(parents=True, exist_ok=True)
+        self._config = config
+        self._templates = templates
+
+    def _write_yaml(self, headers: HeaderFile):
```

Comment:
```suggestion
    def _write_yaml(self, headers: HeaderFile) -> None:
```
Similar here about the plural consistency.